Correct indentation in ACL comparator

bewing opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Change indentation of ACL present check in _state_replaced to only try to add a missing wanted ACL after processing all have ACLs.

Possibly fixes #512 

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->

- Bugfix Pull Request


##### COMPONENT NAME
`arista.eos.eos_acls`

##### ADDITIONAL INFORMATION

bewing commented:
@Ruchip16 is there anything needed from my side to move this PR forward?

Ruchip16 commented:
> @Ruchip16 is there anything needed from my side to move this PR forward?

hey thank you for raising the PR, can you kindly add integration tests as well to validate the change on device?

bewing commented:
I will take a look at the integration test framework this week, thanks!

bewing commented:
After a few hours screwing around, I was able to generate a Containerlab cEOS image that I can run integration tests on.  It appears that this change does affect one of the integration tests:

```nowrap
TASK [eos_acls : ansible.builtin.debug] ****************************************
ok: [clab-ceos-ceos1] => {
    "msg": "Start eos_acls replaced integration tests ansible_connection=ansible.netcommon.network_cli"
}

TASK [eos_acls : ansible.builtin.include_tasks] ********************************
included: /home/bewing/projects/network/naac/ansible_collections/arista/eos/tests/output/.tmp/integration/eos_acls-1h0nrzag-ÅÑŚÌβŁÈ/tests/integration/targets/eos_acls/tests/common/_populate.yaml for clab-ceos-ceos1

TASK [eos_acls : Setup] ********************************************************
changed: [clab-ceos-ceos1]

TASK [eos_acls : ansible.builtin.set_fact] *************************************
ok: [clab-ceos-ceos1]

TASK [eos_acls : replace attributes with given acls.] **************************
changed: [clab-ceos-ceos1]

TASK [eos_acls : arista.eos.eos_facts] *****************************************
ok: [clab-ceos-ceos1]

TASK [eos_acls : ansible.builtin.assert] ***************************************
ok: [clab-ceos-ceos1] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [eos_acls : Idempotency check] ********************************************
ok: [clab-ceos-ceos1]

TASK [eos_acls : arista.eos.eos_facts] *****************************************
ok: [clab-ceos-ceos1]

TASK [eos_acls : ansible.builtin.assert] ***************************************
fatal: [clab-ceos-ceos1]: FAILED! => {
    "assertion": "result.changed == true",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}

TASK [eos_acls : ansible.builtin.include_tasks] ********************************
included: /home/bewing/projects/network/naac/ansible_collections/arista/eos/tests/output/.tmp/integration/eos_acls-1h0nrzag-ÅÑŚÌβŁÈ/tests/integration/targets/eos_acls/tests/common/_remove_config.yaml for clab-ceos-ceos1

TASK [eos_acls : Setup] ********************************************************
changed: [clab-ceos-ceos1]

PLAY RECAP *********************************************************************
clab-ceos-ceos1            : ok=27   changed=6    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

NOTICE: To resume at this test target, use the option: --start-at eos_acls
FATAL: Command "ansible-playbook eos_acls-pasd2sq0.yml -i inventory.networking" returned exit status 2.
```

I'll have to check what the test is supposed to test, and if it was just written to pass incorrectly prior to this change, or if there is a breaking change included in this PR

bewing commented:
After digging into the newly failing integration test, it appears that an idempotency check was incorrectly written to pass when it shouldn't -- if we re-run the same "replaced" operation with the same config as before, we should send no commands, and there should be no changes.

Updated test to reflect.

Ruchip16 commented:
> After a few hours screwing around, I was able to generate a Containerlab cEOS image that I can run integration tests on. It appears that this change does affect one of the integration tests:
> 
> ```
> TASK [eos_acls : ansible.builtin.debug] ****************************************
> ok: [clab-ceos-ceos1] => {
>     "msg": "Start eos_acls replaced integration tests ansible_connection=ansible.netcommon.network_cli"
> }
> 
> TASK [eos_acls : ansible.builtin.include_tasks] ********************************
> included: /home/bewing/projects/network/naac/ansible_collections/arista/eos/tests/output/.tmp/integration/eos_acls-1h0nrzag-ÅÑŚÌβŁÈ/tests/integration/targets/eos_acls/tests/common/_populate.yaml for clab-ceos-ceos1
> 
> TASK [eos_acls : Setup] ********************************************************
> changed: [clab-ceos-ceos1]
> 
> TASK [eos_acls : ansible.builtin.set_fact] *************************************
> ok: [clab-ceos-ceos1]
> 
> TASK [eos_acls : replace attributes with given acls.] **************************
> changed: [clab-ceos-ceos1]
> 
> TASK [eos_acls : arista.eos.eos_facts] *****************************************
> ok: [clab-ceos-ceos1]
> 
> TASK [eos_acls : ansible.builtin.assert] ***************************************
> ok: [clab-ceos-ceos1] => {
>     "changed": false,
>     "msg": "All assertions passed"
> }
> 
> TASK [eos_acls : Idempotency check] ********************************************
> ok: [clab-ceos-ceos1]
> 
> TASK [eos_acls : arista.eos.eos_facts] *****************************************
> ok: [clab-ceos-ceos1]
> 
> TASK [eos_acls : ansible.builtin.assert] ***************************************
> fatal: [clab-ceos-ceos1]: FAILED! => {
>     "assertion": "result.changed == true",
>     "changed": false,
>     "evaluated_to": false,
>     "msg": "Assertion failed"
> }
> 
> TASK [eos_acls : ansible.builtin.include_tasks] ********************************
> included: /home/bewing/projects/network/naac/ansible_collections/arista/eos/tests/output/.tmp/integration/eos_acls-1h0nrzag-ÅÑŚÌβŁÈ/tests/integration/targets/eos_acls/tests/common/_remove_config.yaml for clab-ceos-ceos1
> 
> TASK [eos_acls : Setup] ********************************************************
> changed: [clab-ceos-ceos1]
> 
> PLAY RECAP *********************************************************************
> clab-ceos-ceos1            : ok=27   changed=6    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
> 
> NOTICE: To resume at this test target, use the option: --start-at eos_acls
> FATAL: Command "ansible-playbook eos_acls-pasd2sq0.yml -i inventory.networking" returned exit status 2.
> ```
> 
> I'll have to check what the test is supposed to test, and if it was just written to pass incorrectly prior to this change, or if there is a breaking change included in this PR

kudos that you spawned up an instance to run the integration tests on and yep generally integration tests might break if the changes in PR are not validated hence its a good way to add integration tests and test them on device such that we are sure that we aren't introducing any breaking changes or infact changes that could fail the CI, thanks again and awesome work 💯

Ruchip16 reviewed (approved):
looks good to me now, thanks for your contribution on this one 💯
